x-pack/plugins/security_solution/public/management/components/artifact_list_page/artifact_list_page.tsx

```diff
@@ -187,9 +198,11 @@ export const ArtifactListPage = memo<ArtifactListPageProps>(
     );
 
     const handleArtifactDeleteModalOnSuccess = useCallback(() => {
-      setSelectedItemForDelete(undefined);
-      refetchListData();
-    }, [refetchListData]);
+      if (isMounted) {
+        setSelectedItemForDelete(undefined);
+        refetchListData();
```

Comment:
The rect query  setting `refetchOnMount: true` should do this, no?

Reply:
I'm not sure what `refetchOnMount` does. My guess (based on the name of the option) is that for cases where you might mount/un-mount the same component (ex. the JSX was cached somewhere), the data will be re-fetched on every `mount`.